tools/project_manager/add_files.php

```diff
@@ -621,8 +609,7 @@ public function _get_action(string $base, string $toi, array $db_exts, array $sr
             $same = (
                 is_file($p_file)
                 &&
-                shell_exec("cmp " . escapeshellarg($src_file)
-                    . " " . escapeshellarg($p_file)) == ''
+                md5_file($src_file) == md5_file($p_file)
```

Comment:
For my understanding, we are not concerned about collisions on those files as uploading require the PM permission, which is only for a few selected users, right?

Reply:
Exactly correct -- we don't need a bytelevel comparison and low probability of collisions in the md5 space are acceptable. Also, the failure scenario is that we determine they are not the same and the project file is replaced by a user who has permissions to do that anyway.